libfwupdplugin/fu-device.c

```diff
@@ -1493,9 +1485,8 @@ fu_device_set_logical_id (FuDevice *self, const gchar *logical_id)
 const gchar *
 fu_device_get_protocol (FuDevice *self)
 {
-	FuDevicePrivate *priv = GET_PRIVATE (self);
 	g_return_val_if_fail (FU_IS_DEVICE (self), NULL);
-	return priv->protocol;
+	return fwupd_device_get_protocol (FWUPD_DEVICE (self));
 }
```

Comment:
Maybe this whole function should move to `fu-device.h` as a #define to `fwupd_device_get_protocol`?

Reply:
No can do, it's a symbol in fwupdplugin.map now. I also think it might be the kind of thing we want to validate in the future, e.g. having an enumerated list of protocols, or checking that the protocol doesn't change on the FuDevice, for instance.

Reply:
Ah right, okay.